lib/improver/utilities/temporal.py

```diff
@@ -525,3 +526,51 @@ def process(self, cube_t0, cube_t1):
             interpolated_cubes.append(single_time)
 
         return interpolated_cubes
+
+
+def unify_forecast_reference_time(cubes, cycletime):
+    """Function to unify the forecast_reference_time across the input cubes
+    provided. The cycletime specified is used as the forecast_reference_time.
+
+    Args:
+        cubes (iris.cube.CubeList):
+            Cubes that will have their forecast_reference_time unified.
+        cycletime (datetime.datetime):
+            Datetime for the cycletime that will be used to replace the
+            forecast_reference_time on the individual cubes.
+
+    Returns:
+        result_cubes (iris.cube.CubeList):
+            Cubes that have had their forecast_reference_time unified.
+
+    """
+    if isinstance(cubes, iris.cube.Cube):
+        cubes = iris.cube.CubeList([cubes])
+
+    result_cubes = iris.cube.CubeList([])
+    for cube in cubes:
+        frt_units = cube.coord('forecast_reference_time').units
+        frt_points = [frt_units.date2num(cycletime)]
+        cube.replace_coord(
+            build_coordinate(
+                frt_points,
+                standard_name="forecast_reference_time",
+                bounds=None,
+                template_coord=cube.coord('forecast_reference_time')))
+
+        # If a forecast period coordinate already exists on a cube, replace
+        # this coordinate, otherwise create a new coordinate.
+        fp_units = "seconds"
+        if cube.coords("forecast_period"):
+            fp_units = cube.coord("forecast_period").units
+            cube.replace_coord(
+                forecast_period_coord(
+                    cube, force_lead_time_calculation=True,
+                    result_units=fp_units))
+        else:
+            fp_coord = forecast_period_coord(
+                cube, force_lead_time_calculation=True, result_units=fp_units)
```

Comment:
This function is called in both if/else blocks. Refactor to run it before instead and use result twice.